action.yaml

```
@@ -31,6 +35,14 @@ inputs:
     description: Hash of the latest commit in the Pull Request.
     required: false
     default: ${{ github.event.pull_request.head.sha }}
```

Wouldn't it be enough to update the `default` here without doing any other changes? Something like this (not sure about the real syntax):

```suggestion
    default: ${{ github.event.pull_request.head.sha || github.event.merge_group.head_sha }}
```

---

don't think we need update the default here.. added another param `merge-group-head` which will be used in merge queue

---

haven't consider `push` events, do we need to consider that as well?

---

Hmm, from my point of view, it looks like the behavior of the Checkton action doesn't need to change at all depending on whether it runs on a regular PR or a PR in the merge queue. The only thing that needs to change is how the action figures out the HEAD and BASE commits.

In fact, I'd say the action already supports the merge queue, you would just have to pass a different value for the `pull-request-base` and `pull-request-head` params. But it would make sense to me to fix the default value for those params so that the user doesn't need to care about that

---

Yes agree as you mention it will work. I see there are two approaches here:
- 1. if we want to use the checkton in current state without changing anything, we have to determine it is `pull_request` or `merge_group` event on the users workflow itself, like I tried in [this PR](https://github.com/susdas-qe/test-merge-queue/pull/9/files),
- 2. if we want to offload the logic from the users workflow and move it to the action implementation, users of the action might not have to worry about in which workflow he is using the action in.
This PR is trying to implement the second approach.

---

That makes sense to me, but I think all we need to is change the `default`. Testing in https://github.com/susdas-qe/test-merge-queue/pull/10

---

Updated the `default` now.